php_phongo.c

```diff
@@ -1886,6 +1909,25 @@ PHP_MINIT_FUNCTION(mongodb)
 	phongo_std_object_handlers.get_closure          = NULL;
 	*/
 
+	/* Initialize zend_class_entry dependencies.
+	 *
+	 * Although DateTimeImmutable was introduced in PHP 5.5.0,
+	 * php_date_get_immutable_ce() is not available in PHP versions before
+	 * 5.5.24 and 5.6.8.
+	 *
+	 * Although JsonSerializable was introduced in PHP 5.4.0,
+	 * php_json_serializable_ce is not exported in PHP versions before 5.4.26
+	 * and 5.5.10. For later PHP versions, looking up the class manually also
+	 * helps with distros that disable LTDL_LAZY for dlopen() (e.g. Fedora).
+	 */
+	php_phongo_date_immutable_ce = php_phongo_fetch_internal_class(ZEND_STRL("datetimeimmutable") TSRMLS_CC);
+	php_phongo_json_serializable_ce = php_phongo_fetch_internal_class(ZEND_STRL("jsonserializable") TSRMLS_CC);
+
+	if (php_phongo_json_serializable_ce == NULL) {
+		zend_error(E_ERROR, "JsonSerializable class not defined. Please ensure that the json extension is loaded before the mongodb extension.");
```

Comment:
@derickr I think you cannot trigger this error in MINIT which is never if dependencies are not satisfied (see my test results above)